BE/src/main/java/louie/dong/airbnb/accommodation/AccommodationService.java

```diff
@@ -0,0 +1,116 @@
+import louie.dong.airbnb.accommodation.dto.AccommodationSearchResponse;
+import louie.dong.airbnb.wishlist.WishRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class AccommodationService {
+
+	private final AccommodationRepository accommodationRepository;
+	private final WishRepository wishRepository;
+
+	public AccommodationPriceResponse findPrices(String country) {
+		List<Integer> prices = accommodationRepository.findByAccommodationPrices(country);
+		return new AccommodationPriceResponse(calculateAverage(prices), prices);
+	}
+
+	public AccommodationDetailResponse findById(Long id) {
+		Accommodation accommodation = getAccommodation(id);
+		boolean wish = wishRepository.existsByAccommodationId(id);
+		return new AccommodationDetailResponse(accommodation, wish);
+	}
+
+	public AccommodationDetailPriceResponse findDetailPrice(Long id,
+		AccommodationDetailPriceRequest accommodationDetailPriceRequest) {
+		Accommodation accommodation = getAccommodation(id);
+		LocalDate checkIn = accommodationDetailPriceRequest.getCheckIn();
+		LocalDate checkOut = accommodationDetailPriceRequest.getCheckOut();
+
+		int date = (int) checkIn.until(checkOut, ChronoUnit.DAYS);
+		int totalPrice = accommodation.getPrice() * date;
+		double discountRate = getDiscountRate(date) * 0.01;
+		int discountPrice = (int) (totalPrice * discountRate);
+		int finalPrice = totalPrice - discountPrice + accommodation.getCleaningFee()
+			+ accommodation.getServiceFee() + accommodation.getAccommodationFee();
+
+		return new AccommodationDetailPriceResponse(accommodation.getPrice(), date, totalPrice,
+			WEEKLY.getDiscountRate(), discountPrice, accommodation.getCleaningFee(),
+			accommodation.getServiceFee(), accommodation.getAccommodationFee(), finalPrice);
+	}
+
+	public AccommodationSearchResponse findAccommodations(
+		AccommodationSearchRequest accommodationSearchRequest) {
+		LocalDate checkIn = accommodationSearchRequest.getCheckIn();
+		LocalDate checkOut = accommodationSearchRequest.getCheckOut();
+
+		List<Accommodation> accommodations = accommodationRepository.searchAccommodations(
+			accommodationSearchRequest.getCountry(),
+			LocalDateTime.of(checkIn, LocalTime.of(0, 0)),
+			LocalDateTime.of(checkOut, LocalTime.of(0, 0)),
+			accommodationSearchRequest.getMinPrice(), accommodationSearchRequest.getMaxPrice(),
+			accommodationSearchRequest.getGuestCount());
+
+		List<AccommodationResponse> accommodationResponses = createAccommodationResponses(
+			accommodations, checkIn, checkOut);
+		return new AccommodationSearchResponse(accommodations.size(), accommodationResponses);
+	}
+
+	private List<AccommodationResponse> createAccommodationResponses(
+		List<Accommodation> accommodations, LocalDate checkIn, LocalDate checkOut) {
+		int stayNight = (int) ChronoUnit.DAYS.between(checkIn, checkOut);
```

Comment:
dto 생성 동작은 dto가 가지고 있으면 좋을듯 합니다.
dto 리스트 생성에 복잡한 로직이 포함된다면 일급컬렉션을 사용해 보는것도 좋을듯 합니다.

Reply:
메소드 파라미터 개수를 줄일 수 있을것 같네요
```suggestion
	private List<AccommodationResponse> createAccommodationResponses(
		List<Accommodation> accommodations, int nights) {
```

Reply:
Service에 dto를 변환하는 로직이 많은 것 같아서 고민이였는데 dto를 생성할 때 중요한 데이터 정도만 Service 계층에서 만들어주고 나머지는 dto의 생성자에서 처리해줘도 좋겠네요!
감사합니다 왕민 🤩